src/runtime_src/core/tools/common/XBHelpMenus.cpp

```diff
@@ -266,31 +266,34 @@ XBUtilities::produce_reports( xrt_core::device_collection devices,
         is_recovery = false;
       }
 
+      // Process the tests that require a device
+      // If the device is either of the following, most tests cannot be completed fully:
+      // 1. Is in factory mode and is not in recovery mode
+      // 2. Is not ready and is not in recovery mode
+      if((is_mfg || !is_ready) && !is_recovery) {

```

Comment:
This was the original, I converted the comment to reflect what is actually going on vs what were are trying to avoid. Although we are not skipping the tests anymore, rather, just outputting an error message.
`//if the device is either of the following, continue to create reports:

        // 1. not in factory mode and is ready to use
        // 2. is in recovery mode
        if((!is_mfg && !is_ready) && !is_recovery)
          continue;`